Neos.Flow/Classes/Mvc/Controller/AbstractController.php

```diff
@@ -38,16 +40,28 @@
 abstract class AbstractController implements ControllerInterface
 {
     /**
+     * @deprecated with Flow 8.3 - use $this->actionUriBuilder instead
      * @var UriBuilder
      */
     protected $uriBuilder;
 
+    /**
+     * @var ActionUriBuilder|null
+     */
+    protected ?ActionUriBuilder $actionUriBuilder;
```

Comment:
should probably not null able so people dont have to check, and psalm or phpstan cant complain

Reply:
yes but our ugly test would fail, as initializeController will be called with actionUriBuilderFactory as a non functioning mock currently ... so we assing null at some point

Reply:
It has to be nullable because it's not set at construction time unfortunately

Reply:
hmm this works?
https://onlinephp.io?s=dY-xDsIwDERnIuUfrCoDDF2ArdBu_AeEtIlUxVXiiAH13wlNqkZIeLw7v7Mv3aQnzjiT4917uCFy9uYM4kwOSUlST_DkjB1A9IhN8nIiPEYjoQ9WkkELXlEE7Iv4IQUzcSdIG1-30YBriZtL6n_-kPg_UKcoOAsbe4PO3884E6nQqhcs-80q1m0-uTqeT1XSldQIyVz7Fv0D&v=8.2.4

the only rule is, that it can only be initialized once and might not be accessed (except isset and ??)